libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2514,22 +2514,31 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		mstore(<pos>, <shl28>(<funSel>))
 		let <end> := <encodeArgs>(add(<pos>, 4) <argumentString>)
 
-		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <reservedReturnSize>)
+		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <estimatedReturnSize>)
 		<?noTryCall>
 			if iszero(<success>) { <forwardingRevert>() }
 		</noTryCall>
 		<?+retVars> let <retVars> </+retVars>
 		if <success> {
+			let <returnDataSize>
 			<?dynamicReturnSize>
 				// copy dynamic return data out
-				returndatacopy(<pos>, 0, returndatasize())
+				<returnDataSize> := returndatasize()
+				returndatacopy(<pos>, 0, <returnDataSize>)
+			<!dynamicReturnSize>
+				<returnDataSize> := <estimatedReturnSize>
```

Comment:
You don't have to separately declare the `returnDataSize` variable as if it was an `if` branch. I think it's clearer (and shorter) this way:
```suggestion
				let <returnDataSize> := returndatasize()
				returndatacopy(<pos>, 0, <returnDataSize>)
			<!dynamicReturnSize>
				let <returnDataSize> := <estimatedReturnSize>
```